storage/replica_raftstorage.go

```diff
@@ -141,6 +141,9 @@ func (r *Replica) LastIndex() (uint64, error) {
 // current entry. This includes both entries that have been compacted away
 // and the dummy entries that make up the starting point of an empty log.
 func (r *Replica) raftTruncatedState() (proto.RaftTruncatedState, error) {
+	if r.truncatedState != nil {
```

Comment:
I think it is unsafe to examine `r.truncatedState` directly like this. We must always go through `atomic.LoadPointer`.